scripts/prune_ingestion_sequence.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python
+"""
+Script for deleting items for a given provider and sequence number.
+
+Unless the argument 'no_dry_run' is given as the last argument, this will
+operate in a non-destructive mode and just print out the records it would
+delete for safety. Otherwise, it will delete them.
+
+NOTE: There is a bug with Elasticsearch actually processing all the DELETE
+requests, at least on my VM. This script may need to be run more than once
+depending on how many records there are to delete.
+
+Usage:
+    $ python prune_ingestion_sequence.py ingestion_document_id\
+     sequence_number no_dry_run|anything
+
+"""
+
+import sys
+import argparse
+from amara.thirdparty import json
+import ConfigParser
+import requests
+import urllib
+
+
+def define_arguments():
+    """Defines command line arguments for the current script"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("profile_path",
+                        help="The path to the profile to be processed")
+    parser.add_argument("ingestion_sequence",
+                        help="The ingestionSequence number to remove")
+    parser.add_argument("dry_run",
+                        help="Only deletes the records if this argument is "
+                             "'no_dry_run'. Otherwise, prints the potentially "
+                             "deleted ids.")
+
+    return parser
+
+
+def get_provider_id(profile_path):
+    with open(profile_path, "r") as f:
+        try:
+            profile = json.load(f)
+            return profile['contributor']['@id']
+        except Exception, err:
+            print "Error, could not load profile in %s: %s" % (__name__, err)
+            return None
+
+
+def get_elasticsearch_url():
+    config = ConfigParser.ConfigParser(
+        {"LogLevel": "INFO", "SyncQAViews": "True"}
+    )
+    with open("akara.ini") as f:
+        try:
+            config.readfp(f)
+            return config.get("Elasticsearch", "Index")
+        except Exception, err:
+            print "Error, couldn't read elasticsearch index from config: %s" \
+                  % err
+            return None
+
+
+def handle_destructive_results(base_url, results):
+    for hit in results:
+        _type = str(hit["_type"])
+        _id = urllib.quote_plus(str(hit["_id"]))
+        url = "%s/%s/%s" % (base_url, _type, _id)
+        response = requests.delete(url)
```

Comment:
Since this is a one-off script, i was just going to let errors bubble. I can't really do anything about an exception or unspecified behavior aside from throw a new, more verbose exception, which I guess I'll do.